client.go

```diff
@@ -163,31 +184,49 @@ func (c *Client) request(stream string) (*http.Response, error) {
 func (c *Client) processEvent(msg []byte) *Event {
 	var e Event
 
-	switch h := msg; {
-	case bytes.Contains(h, headerID):
-		e.ID = trimHeader(len(headerID), msg)
-	case bytes.Contains(h, headerData):
-		e.Data = trimHeader(len(headerData), msg)
-	case bytes.Contains(h, headerEvent):
-		e.Event = trimHeader(len(headerEvent), msg)
-	case bytes.Contains(h, headerError):
-		e.Error = trimHeader(len(headerError), msg)
-	default:
-		return nil
+	// Normalize the crlf to lf to make it easier to split the lines.
+	bytes.Replace(msg, []byte("\n\r"), []byte("\n"), -1)
+	// Split the line by "\n" or "\r", per the spec.
+	for _, line := range bytes.FieldsFunc(msg, func(r rune) bool { return r == '\n' || r == '\r' }) {
+		switch {
+		case bytes.HasPrefix(line, headerID):
+			e.ID = trimHeader(len(headerID), line)
+		case bytes.HasPrefix(line, headerData):
+			// The spec allows for multiple data fields per event, concatenated them with "\n".
+			e.Data = append(append(trimHeader(len(headerData), line), e.Data[:]...), byte('\n'))
+		// The spec says that a line that simply contains the string "data" should be treated as a data field with an empty body.
+		case bytes.Equal(line, bytes.TrimSuffix(headerData, []byte(":"))):
+			e.Data = append(e.Data, byte('\n'))
+		case bytes.HasPrefix(line, headerEvent):
+			e.Event = trimHeader(len(headerEvent), line)
+		case bytes.HasPrefix(line, headerRetry):
+			e.Retry = trimHeader(len(headerRetry), line)
+		default:
+			return nil
+		}
 	}
 
-	if len(e.Data) > 0 && c.EncodingBase64 {
-		buf := make([]byte, base64.StdEncoding.DecodedLen(len(e.Data)))
+	if len(e.Data) > 0 {
```

Comment:
It wasn't clear to me from the spec whether or not I should trim e.Data before checking it's length... What do you think? (I'm kinda leaning towards trimming then checking...)

Here's the relevant passages from the spec:

Reading from the stream:
```
If the field name is "data"
Append the field value to the data buffer, then append a single U+000A LINE FEED (LF) character to the data buffer.
```
Processing the event after reading from the stream:
```
If the data buffer is an empty string, set the data buffer and the event type buffer to the empty string and return.

If the data buffer's last character is a U+000A LINE FEED (LF) character, then remove the last character from the data buffer.
```

Reply:
It sounds to me like if the last character is LF, then an event should be sent, albeit with an empty data field.

Reply:
Actually, I reread and agree. Change pushed.